main.nf

```diff
@@ -431,12 +193,36 @@ process calculateStats {
         mode: "copy"
 
     input:
-        path outfile_basename from final_output_vcf
+        path outfile_basename
 
     output:
         path "${outfile_basename}.stats"
-        
+
     """
     bcftools stats ${outfile_basename} > ${outfile_basename}.stats
     """
 }
+
+
+workflow {
+    main:
+        prepare_old_genome(params.oldgenome)
+        prepare_new_genome(params.newgenome)
+        uncompressInputVCF(params.vcffile)
+        StoreVCFHeader(uncompressInputVCF.out.vcf_file)
+        process_split_reads(
+            uncompressInputVCF.out.vcf_file,
+            params.oldgenome,
+            prepare_old_genome.out.genome_fai,
+            prepare_old_genome.out.genome_chrom_sizes,
+            params.newgenome,
+            prepare_new_genome.out.genome_fai
+        )
+        sortVCF(process_split_reads.out.variants_remapped)
+        buildHeader(sortVCF.out.variants_remapped_sorted, StoreVCFHeader.out.vcf_header)
+        mergeHeaderAndContent(buildHeader.out.final_header, sortVCF.out.variants_remapped_sorted)
+        normalise(mergeHeaderAndContent.out.final_vcf_with_header, params.newgenome)
+        calculateStats(normalise.out.final_output_vcf)
```

Comment:
Just wanted to clarify, since I didn't work with DSL 2: the steps will still execute in parallel, right? So this is not a _sequence_ of operations, but rather a list of recipes to be executed asynchronously once all the inputs become available?

Reply:
I think the same rule apply on parallelism as before. All independent tasks are executed in parallel and interdependent tasks are in serial.